### ISSUE TITLE
[BUG] `no-duplicate-attrs` false positive with Go templates

### ISSUE THREAD

**Issue by AdamVig** [yeonjuan/html-eslint#409]
**Describe the bug**
I'm using a Go template conditional to swap class names and the second `class` attribute (in the "else" clause) is incorrectly flagged by this plugin:
```
The attribute 'class' is duplicated. eslint(@html-eslint/no-duplicate-attrs)
```
I can't find a way to disable it, either, since HTML doesn't allow comments within tags and the violation occurs a few lines below the opening tag. This `eslint-disable` pragma has no effect:
```handlebars
<!-- eslint-disable-next-line --> <-- ineffectual pragma
<button
  {{if eq .Data.UiState.CurrentFilter "A"}}
    class="filter-button active"
  {{else}}
    class="filter-button" <-- line with violation
  {{end}}
></button>
```

**To Reproduce**
Use this code:
```handlebars
<button
  {{if eq .Data.UiState.CurrentFilter "A"}}
    class="filter-button active"
  {{else}}
    class="filter-button"
  {{end}}
></button>
```
**Expected behavior**
I'm not sure. Obviously this plugin can't be expected to parse template expressions, so the next best option would be to somehow make `eslint-disable-next-line` work for this scenario, i.e. by considering the "next line" to include the entire opening tag, even if it's split across multiple lines:
```handlebars
<!-- eslint-disable-next-line --> <-- start of scope
<button
  {{if eq .Data.UiState.CurrentFilter "A"}}
    class="filter-button active"
  {{else}}
    class="filter-button"
  {{end}}
>
<---- end of eslint-disable-next-line's scope ---->
</button>
```

**Comment by yeonjuan** [comment]
Hi Thank you for the report.  Yes, currently, we do not know whether attributes are used conditionally by the template engine. I think we need to improve the parts related to the template engine. I think it would be good if the plugin could recognize that it is being rendered conditionally.

**Comment by Simbiat** [comment]
I hit the same issue in Twig, too, using something like:
```twig
    <p>
        {% if check_result is same as(null) %}
            <span id="acc_check_result"></span>
        {% elseif check_result is same as(false) %}
            <span id="acc_check_result" class="failure">Неверный формат БИКа или счёта</span>
        {% elseif check_result is same as(true) %}
            <span id="acc_check_result" class="success">Правильное ключевание</span>
        {% else %}
            <span id="acc_check_result" class="failure">Неверное ключевание. Ожидаемый ключ: {{ check_result }} ({{ first_half }}<span class="success">{{ check_result }}</span>{{ second_half }})</span>
        {% endif %}
    </p>
```
Similar thing can happen with duplicate attributes rule, too. I asked Claude if there is a good way to address this, and it created this new utility module (to be placed in `rules/utils`) [twig-branches.js](https://github.com/user-attachments/files/29440369/twig-branches.js) and updated these rules (I think only these 3 make sense):

- [no-duplicate-attrs.js](https://github.com/user-attachments/files/29440377/no-duplicate-attrs.js)
- [no-duplicate-id.js](https://github.com/user-attachments/files/29440376/no-duplicate-id.js)
- [no-duplicate-in-head.js](https://github.com/user-attachments/files/29440378/no-duplicate-in-head.js)

I manually tested mainly `no-duplicate-id.js` and it does seem to work: the above sample no longer gets highlighted. I also tried something like
```twig
{% if layer1 %}
    {% if layer2 %}
        <span id="acc_check_result2"></span>
    {% else %}
        <span id="acc_check_result2"></span>
    {% endif %}
{% else %}
    <span id="acc_check_result2"></span>
{% endif %}
```
Also no highlights, but as soon as I add one more line like:
```twig
{% if layer1 %}
    {% if layer2 %}
        <span id="acc_check_result2"></span>
    {% else %}
        <span id="acc_check_result2"></span>
    {% endif %}
{% else %}
    <span id="acc_check_result2"></span>
{% endif %}
<span id="acc_check_result2"></span>
```
All duplicates are highlighted.

With attributes I tested
```twig
<img crossorigin="anonymous" loading="lazy" decoding="async" class="link_icon gallery_zoom" alt=""
      {% if freecompany.grand_company == "Maelstrom" %}
          src="/assets/images/fftracker/cities/4.webp"
      {% elseif freecompany.grand_company == "Immortal Flames" %}
          src="/assets/images/fftracker/cities/5.webp"
      {% else %}
          src="/assets/images/fftracker/cities/2.webp"
      {% endif %}
       width="1024" height="1024">
```
and that also works fine.

This is obviously not a universal solution, but perhaps `SyntaxConfig` can be expanded? It already supports `isComment` flag, so maybe `isCondition` can be added as well as `else` key, that would accept an array of strings, and then some utility based on `twig-branches.js` above uses that, if available? `TWIG` constant would be something like
```javascript
const TWIG = [
  { open: "{{", close: "}}" },
  { open: "{%", close: "%}", isCondition: true, else: ["{% else %}", "{% elseif %}"] },
  { open: "{#", close: "#}", isComment: true },
];
```
This won't help with ERB at all due to how ambiguous its `end` tag is, but Handlebars/Nunchucks and other similar templates should work, if proper config is provided. Although, to be honest, I am not sure how reliable the regex from `twig-branches.js` is, maybe that's a bad approach altogether. But if this is an acceptable approach (as well as use of AI for helping with this) and we agree on a way to expand `SyntaxConfig`, I can create a PR for this.

**Comment by yeonjuan** [comment]
> This won't help with ERB at all due to how ambiguous its end tag is, but Handlebars/Nunchucks and other similar templates should work, if proper config is provided. Although, to be honest, I am not sure how reliable the regex from twig-branches.js is, maybe that's a bad approach altogether. But if this is an acceptable approach (as well as use of AI for helping with this) and we agree on a way to expand SyntaxConfig, I can create a PR for this.

@Simbiat  Thanks for looking into this! I'm actually considering a more fundamental solution to this issue.
Rather than relying on SyntaxConfig settings, I think it would be better to extend the parser so that it can produce a more accurate AST.
I haven't had the chance to invest much time in this yet, but I think the right approach would be either to develop a more specialized parser or to find an existing one that better fits our needs and make it compatible with our AST.

**Comment by Simbiat** [comment]
you mean like a separate parser to generate a unique tree for each individual template engine? But, as far as I know, you can, technically customize the characters used in their syntax (at least I know in Twig it's possible, but never used that). I think that alone means that you can't specialize the parser, or at least you would not avoid some sort of SyntaxConfig.

Also, because of the conditions, you would then need to have multiple ASTs, unless you want an AST which is not really the same as HTML's AST. Then if you have multiple ASTs you will need a way to somehow differentiate them when outputting the results of the rule, so that human can understand what's wrong and what need to be parsed.

Not saying it's impossible, and I can agree that a specialized parser may be better, because it can help catch more niche bugs in HTML (and maybe in the template syntax itself). But considering that we, technically, just need a way to _exclude_ some duplicates based on conditions, this may be overcomplicating things a bit.
